app/services/org/create_created_plan_service.rb

```diff
@@ -6,24 +6,54 @@ def call(org = nil)
 
         orgs.each do |org|
           OrgDateRangeable.split_months_from_creation(org) do |start_date, end_date|
-            create_count_for_date(start_date: start_date, end_date: end_date, org: org)  
+            create_or_update(start_date: start_date, end_date: end_date, org: org)
           end
         end
       end
 
       private
 
+      def users(org)
+        User.where('users.org_id = ?', org.id)
+      end
+
+      def plans(start_date:, end_date: )
+        Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
+      end
+
+      def creator_admon
+        Role.with_access_flags(:creator, :administrator)
+      end
+
       def count_plans(start_date: , end_date: , org:)
-        users = User.where('users.org_id = ?', org.id)
-        plans = Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
-        creator_admon = Role.with_access_flags(:creator, :administrator)
+        users = users(org)
+        plans = plans(start_date: start_date, end_date: end_date)
 
         Role.joins([:plan, :user]).merge(creator_admon).merge(users).merge(plans).select(:plan_id).distinct.count
       end
 
-      def create_count_for_date(start_date:, end_date:, org:)
+      def by_template(start_date:, end_date:, org:)
+        users = users(org)
+        plans = plans(start_date: start_date, end_date: end_date)
+        roleable_plan_ids = Role.joins([:plan, :user]).merge(creator_admon).merge(users).merge(plans).select(:plan_id).distinct
+        template_counts = Plan.where(id: roleable_plan_ids).group(:"template_id").count
+        template_names = Template.where(id: template_counts.keys).pluck(:id, :title)
+        template_names.map do |t|
+          {  name: t[1], count: template_counts[t[0]] }
+        end
+      end
+
+      def create_or_update(start_date:, end_date:, org:)
         count = count_plans(start_date: start_date, end_date: end_date, org: org)
-        StatCreatedPlan.create(date: end_date.to_date, count: count, org_id: org.id)
+        by_template = by_template(start_date: start_date, end_date: end_date, org: org)
+        attrs = { date: end_date.to_date, org_id: org.id, count: count, details: { by_template: by_template }}
+        stat_created_plan = StatCreatedPlan.find_by(date: attrs[:date], org_id: attrs[:org_id])
+
+        if stat_created_plan.present?
```

Comment:
ActiveRecord has a `create_or_update` method you can use instead

Reply:
```create_or_update```, as far as I'm aware is a private method used by #save. Since save is an instance method, it means that we have to find a matching record by date and org_id in order to either create or save the instance therefore that's the reason I used it. Anyway, I have added a uniqueness validation against date/org_id for a stat type (e.g. StatCreatedPlan, StatJoinedUser) in case another piece of code tries to save to the database outside this service. I'm open to other proposals :)